src/common/proto-irc.c

```diff
@@ -1026,6 +1027,9 @@ process_named_msg (session *sess, char *type, char *word[], char *word_eol[],
 		ex[0] = 0;
 		safe_strcpy (nick, word[1], sizeof (nick));
 		ex[0] = '!';
+
+		account = tags_data->account && *tags_data->account ? tags_data->account : "*";
+		userlist_set_account (sess, nick, account);
```

Comment:
If you look at `inbound_account()` what it does is call `userlist_set_account()` for ever session in a server, which I think is what we want.

Reply:
Also looking at it, `userlist_set_account()` always frees and copies the new account, ignoring if its unchanged, so that should be handled somewhere.

Reply:
Fixed.